templates/default/rsyslog.conf.erb

```diff
@@ -30,15 +30,9 @@ $PreserveFQDN <%= node['rsyslog']['preserve_fqdn'] %>
 
 <% if node['rsyslog']['modules'] && !node['rsyslog']['modules'].empty? %>
   <% [*node['rsyslog']['modules']].each do |mod| %>
-  <% case mod -%>
-  <% when 'imuxsock' -%>
-module(load="imuxsock" SysSock.Use="off")
-  <% when 'imjournal' -%>
-module(load="imjournal" UsePid="system" StateFile="imjournal.state")
-  <% else -%>
-module(load="<%= mod %>")
-  <% end %>
-  <% end %>
+module(load="<%= mod %>"<%- node['rsyslog']["#{mod}_directives"]&.each { |k, v| <%= k %>="<%= v %>" })
+  <%- end -%>
+ <%- end -%>
 <% end %>
```

Comment:
so in my testing I could not get this to work with the inline erb tags, which makes sense since we are already in ruby mode for the .each call. I got this to work and render correctly though.
```suggestion
    <% params = []%>
    <% node['rsyslog']["#{mod}_directives"]&.each {|k, v| params.push("#{k}=\"#{v}\"")} %>
module(load="<%= mod %>" <%= params.join(' ') -%>) 
  <%- end -%>
<% end %>
```

Reply:
That generated template output like
```
* template[/tmp/rsyslog.test] action create
           - update content in file /tmp/rsyslog.test from ab54d3 to 19e466
           --- /tmp/rsyslog.test	2024-12-17 23:46:46.541116708 +0000
           +++ /tmp/.chef-rsyslog20241217-2486-od1f1k.test	2024-12-17 23:59:22.018483969 +0000
           @@ -23,8 +23,8 @@
            #### MODULES ####
            #################
            
           -module(load="imuxsock") 
           -module(load="imjournal") 
           +module(load="imuxsock" SysSock.Use="off") 
           +module(load="imjournal" UsePid="system" StateFile="imjournal.state") 
            
            
            ###########################
```